### ISSUE TITLE
Non-ASCII subtitles

### ISSUE THREAD

**Issue by corna** [Sandmann79/xbmc#80]
Hi,

while trying to play a video this plugin fails with this error:

```
NOTICE: [Amazon VOD] Convert English [CC] Subtitle
NOTICE: [Amazon VOD] Srtfile: /storage/.kodi/temp/English [CC].srt
NOTICE: [Amazon VOD] getURL: https://___.dfxp
NOTICE: [Amazon VOD] Convert Nederlands Subtitle
NOTICE: [Amazon VOD] Srtfile: /storage/.kodi/temp/Nederlands.srt
NOTICE: [Amazon VOD] getURL: https://___.dfxp
NOTICE: [Amazon VOD] Convert Русский Subtitle
NOTICE: [Amazon VOD] Srtfile: /storage/.kodi/temp/Русский.srt
ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
 - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
Error Type: <type 'exceptions.UnicodeEncodeError'>
Error Contents: 'ascii' codec can't encode characters in position 20-26: ordinal not in range(128)
Traceback (most recent call last):
  File "/storage/.kodi/addons/plugin.video.amazon-test/default.py", line 3090, in <module>
    PrimeVideo_Browse(None if 'path' not in args else args['path'])
  File "/storage/.kodi/addons/plugin.video.amazon-test/default.py", line 461, in PrimeVideo_Browse
    PlayVideo(node['metadata']['video'], node['metadata']['asin'], '', 0)
  File "/storage/.kodi/addons/plugin.video.amazon-test/default.py", line 1583, in PlayVideo
    playable = IStreamPlayback(asin, name, trailer, isAdult, extern)
  File "/storage/.kodi/addons/plugin.video.amazon-test/default.py", line 1707, in IStreamPlayback
    opt='&titleDecorationScheme=primary-content', dRes=dRes, useCookie=cookie), retmpd=True)
  File "/storage/.kodi/addons/plugin.video.amazon-test/default.py", line 1965, in getStreams
    subUrls = parseSubs(data)
  File "/storage/.kodi/addons/plugin.video.amazon-test/default.py", line 2016, in parseSubs
    with codecs.open(srtfile, 'w', encoding='utf-8') as srt:
  File "/usr/lib/python2.7/codecs.py", line 896, in open
UnicodeEncodeError: 'ascii' codec can't encode characters in position 20-26: ordinal not in range(128)
-->End of Python script error report<--
```

(I've added a print of srtfile for clearness)

The error seems to be related to the russian subtitles, as the srt filename contains non-ASCII characters: `/storage/.kodi/temp/Русский.srt`

**Comment by Varstahl** [comment]
Thanks for reporting it, I'll have a look at it later today or in the weekend. Can you tell me what movie/tv series causes problems, or does it happen with every russian subtitle? I don't believe I encountered this problem ever, yet.

**Comment by corna** [comment]
Grimm, season 1, episode 1. I've not checked other movies, I'll try.

I'm on RPi3 with LibreELEC, the issues mentioned in #62 [have been solved](https://forum.kodi.tv/showthread.php?tid=298461&pid=2701351#pid2701351) (Netflix works perfectly with the newest firmware and libwidevine).

**Comment by Varstahl** [comment]
> Grimm, season 1, episode 1. I've not checked other movies, I'll try.

I'll take a look at it.

> I'm on RPi3 with LibreELEC, the issues mentioned in #62 have been solved (Netflix works perfectly with the newest firmware and libwidevine).

Don't have a RPi around, so I'll just ping @Sandmann79 :)

**Comment by Sandmann79** [comment]
@corna thanks for info. 
I don't have an RPi too, but the user's at kodinerds forum reporting that's working again with newest Milhouse Build.

I think this issue can be solved by wrapping the srtfile in cleanName() method, like at savefile() method done.

**Comment by Varstahl** [comment]
Doesn't look like a naming issue — although I'll admit I wanted to change the names from some time — it looks like an encoding issue.

Since Kodi does not recognize subtitle languages anyway, I was about to check out/rework the subtitle system anyway, this just adds up some more checks to perform on the streams themselves. I'll look at it asap, it's been quite a rough working week.

**Comment by Sandmann79** [comment]
@corna if you want you can try it by yourself. Just change line 2014 at default.py to this:
`srtfile = xbmc.translatePath('special://temp/%s.srt' % cleanName(lang, isfile=False)).decode('utf-8')`

**Comment by corna** [comment]
@Sandmann79 Unfortunately it doesn't solve the issue, it fails with the same error (and `srtfile` is unchanged, still `/storage/.kodi/temp/Русский.srt`)

**Comment by Varstahl** [comment]
I'm on the case.

**Comment by Varstahl** [comment]
Fixed in #82

**Comment by corna** [comment]
Works now, thanks for your help
